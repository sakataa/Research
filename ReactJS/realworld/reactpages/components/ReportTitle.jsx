import React, { Component } from 'react';
import ReactDOM from 'react-dom';

export default class ReportTitle extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="report-title-container">
                <p className="report-title">{this.props.content}</p>
            </div>
        )
    }
}