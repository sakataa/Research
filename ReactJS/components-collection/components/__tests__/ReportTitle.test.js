import React from 'react'
import ReportTitle from '../ReportTitle'
import renderer from 'react-test-renderer'

describe('ReportTitle ', () => {
    it('props content is undefined - component render correctly', () => { 
        const props = {};

        const component = renderer.create(<ReportTitle {...props} />);

        const result = component.toJSON(); 
        
        expect(result).toMatchSnapshot(); 
    });

    it('props content is not empty - component render correctly', () => {
        const props = {
            content: "Title"
        };

        const component = renderer.create(<ReportTitle {...props} />); 

        const result = component.toJSON(); 

        expect(result).toMatchSnapshot(); 
    });
})