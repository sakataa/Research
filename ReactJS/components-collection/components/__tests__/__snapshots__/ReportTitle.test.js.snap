// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTitle  props content is not empty - component render correctly 1`] = `
<div
  className="report-title-container"
>
  <p
    className="report-title"
  >
    Title
  </p>
</div>
`;

exports[`ReportTitle  props content is undefined - component render correctly 1`] = `
<div
  className="report-title-container"
>
  <p
    className="report-title"
  />
</div>
`;
